From 7270383c5a68c34e4b6c3192852072101d76b449 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Przemys=C5=82aw=20Szczepaniak?= <pszczepaniak@google.com>
Date: Mon, 13 Mar 2023 14:54:47 +0000
Subject: [PATCH] Fix OOB Read in convertSubgraphFromHAL

Bug: 269271098
Test: Run the POC
(cherry picked from https://android-review.googlesource.com/q/commit:029b72665f6c7c262255ad7475d4be9fc58d9db0)
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:22a20c01cc01dbb36ecbc68e664a070c4ffad2cf)
Merged-In: I7a5f3780fafd12663693b74988baee6a9188041f
Change-Id: I7a5f3780fafd12663693b74988baee6a9188041f
---
 shim_and_sl/ShimConverter.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/shim_and_sl/ShimConverter.cpp b/shim_and_sl/ShimConverter.cpp
index b8619d0f..94c762a2 100644
--- a/shim_and_sl/ShimConverter.cpp
+++ b/shim_and_sl/ShimConverter.cpp
@@ -51,6 +51,10 @@ ANeuralNetworksModel* convertSubgraphFromHAL(
         size_t subgraphIndex, const std::vector<uint8_t>& copiedOperandValues,
         ErrorStatus* errorStatus) {
     *errorStatus = ErrorStatus::NONE;
+    if (allModels == nullptr || subgraphIndex >= (*allModels).size()) {
+        *errorStatus = ErrorStatus::INVALID_ARGUMENT;
+        return nullptr;
+    }
     if ((*allModels)[subgraphIndex].has_value()) {
         return (*allModels)[subgraphIndex]->getHandle();
     }
-- 
2.43.0.rc1.413.gea7ed67945-goog

